woocommerce/woocommerce-ios
# Bottom Sheet: Products tab sort options

Use case 1: Products tab sort options for #2054 

- [x] ⚠️ update PR base after https://github.com/woocommerce/woocommerce-ios/pull/2155 is merged ⚠️ 

## Changes

- In `BottomSheetViewController`, updated background color for the popover arrow and its view to match its child view controller (that renders the main content)
- Removed `UIAlertController` code for the native action sheet, and moved `ProductsSortOrder+actionSheetTitle` to its own file
- Created `ProductsSortOrderBottomSheetListSelectorDataSource` as the data source for the bottom sheet list selector + basic unit tests
- In `ProductsViewController`, replaced the native action sheet with bottom sheet

## Testing

- Go to the Products tab
- Tap "Sort by" below the WIP banner --> the bottom sheet should show up for selecting the sort order
- Try rotating, dynamic type, on an iPad and the bottom sheet should still look good (note that the iPad sheet doesn't currently fit to content height, I will follow up on this issue)

## Example screencasts/screenshots

dark mode | light mode
-- | --
![bottom-sheet-sort-products-dark](https://user-images.githubusercontent.com/1945542/79726724-d3449980-831d-11ea-9249-8509ca7f6304.gif) | ![bottom-sheet-sort-products-light](https://user-images.githubusercontent.com/1945542/79726741-d9d31100-831d-11ea-888b-e2eec503c059.gif)

iPad (note that the iPad sheet doesn't currently fit to content height, I will follow up on this issue with WPiOS team)
![bottom-sheet-sort-products-ipad](https://user-images.githubusercontent.com/1945542/79726736-d8a1e400-831d-11ea-974f-810c3cdf860d.gif)

dynamic type:

portrait | landscape
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-04-20 at 15 45 05](https://user-images.githubusercontent.com/1945542/79727024-4d751e00-831e-11ea-9161-761bdab5f1a0.png) | ![Simulator Screen Shot - iPhone 11 - 2020-04-20 at 15 45 12](https://user-images.githubusercontent.com/1945542/79727032-50700e80-831e-11ea-8b3b-15f00338462d.png)



Update release notes:
